### ISSUE TITLE
Sprint 2 - Drivetrain - Rio and Kai

### ISSUE THREAD

**Issue by riochan26** [sta-titansrobotics/27294-IntoTheDeep#18]
Goals: 
Finish constructing the drivetrain (without cabling the motors into the driverhub)
Integrate the arm into the drivetrain
Mount DriverHub

**Comment by riochan26** [comment]
October 15th, 2024

After failing the last sprint, Kai and I immediately got back to work, trying to fix the chain. After realizing the chain would be too scuffed if the motors didn't move, we reluctantly decided to follow the REV Robotics guide on creating the drivetrain. After
 starting on it, we were able to quickly progress, finally finalizing the chain design and starting on the mecanum wheels. 

https://github.com/user-attachments/assets/c8228f36-3801-45fb-96b5-0e78f03807f2
Old chain

https://github.com/user-attachments/assets/6520c5ce-8143-46b6-b209-466e182329dc
New Chain and location of motor

**Comment by riochan26** [comment]
October 16th, 2024

After noticing how far our peers were when it came to the drivetrain, Kai and I decided to put more effort into speeding up the process of making the drivetrain. We were able to add the second C-Channel to each of our halves of the drivetrain, and also attach a mecanum wheel to both of them. After that, we decided to test our motors to make sure they work before continuing the project. After testing them, we noticed that the left back wheel was not spinning when we applied power. We hope that it was just a fluke today and it will work fine tomorrow. If it truly does not work, we will have to disassemble the left side of the drivetrain and make sure the motor works as intended before reassembling it.
![image](https://github.com/user-attachments/assets/2aa03c03-489f-4ac4-a5a3-e47d7745c24f)
![image](https://github.com/user-attachments/assets/ef0bd20e-07d7-4867-9e43-c0f08311b9ae)

**Comment by riochan26** [comment]
October 17th, 2024

After further testing of the broken motor, Kai and I realized that something was wrong with the motor. After disassembling it, we learned that the motor was constructed with the incorrect screws, causing it to heat up and not spin functionally. Kai decided to work on preloading a ton of 90 degree brackets to officially bring both sides of the drivetrain together, while I worked on fixing a small detail within our drivetrain.
![image (2)](https://github.com/user-attachments/assets/47c24120-45a9-4bbe-ad29-6fb0bb274d7d)
![image (3)](https://github.com/user-attachments/assets/a317f7b8-5510-4b1a-8fd4-d1a536fd0be7)

**Comment by riochan26** [comment]
October 18th, 2024

With both sides ready, Kai and I decided to finally merge the two sides of the drivetrain to finish the building aspect of the project. After meticulously putting both halves of the drivetrain together, and attaching the remaining two wheels onto the drivetrain. We had completing the building section. With the remaining 5 minutes of class, Kai and I decided to mess around with it and try to watch how the different components interact with each other when moving it. We located a problem where the chained wheels wouldn't move as smooth as we liked, which is where we learned we would have to break down the drivetrain to fix a small issue with the chain.
![image](https://github.com/user-attachments/assets/6d9b8e9e-2aa0-4fa4-a76f-14fb6befb357)

**Comment by riochan26** [comment]
October 22th, 2024

This day was spent breaking down the drivetrain to fix our chain slack issue. Kai and I took apart a small amount of the sides of the drivetrain, and rerouted the chain so that there wouldn't be any slack when switching directions. After reassembling the drivetrain, Kai and I did further testing, and found no imminent issues with our design.
![image](https://github.com/user-attachments/assets/e490a4ea-cad2-4f2f-a9eb-3efe595fbf7f)

**Comment by riochan26** [comment]
October 23th, 2024

After finishing the drivetrain, Thomas also finished a mechanism he was working on, so we decided to integrate the two components together. To do this, the smaller C-Channel on the drivetrain had to be moved to in between the 4 motors. After moving it, integrating the arm took extremely long, and we are still not done yet. We plan on breaking down the drivetrain again to make it easier to attach the arm to the C-Channel.
![image (1)](https://github.com/user-attachments/assets/7d6df80b-9195-4e62-9918-ae873ee8fc2d)

**Comment by riochan26** [comment]
October 24th, 2024

After struggling for a while, Thomas and I were able to integrate the arm with the drivetrain and lock it into place. We then took some time into attaching the driver hub and the battery plate. As we are waiting for the rest of the arm to be finished, Kai and I are planning on either starting the software for the robot's movement, or help the arm team finish their design to put together the robot entirely.
![image (2)](https://github.com/user-attachments/assets/0f9678f7-bb26-4fc3-a3b7-3f6741b15a8d)
